project/src/components/Form.js

```diff
@@ -0,0 +1,265 @@
+                this.props.dispatch(addValue('photo', e.target.files[0], valid));
+                if(valid){
+                    e.target.classList.remove('is-invalid');
+                    e.target.classList.add('is-valid');
+                } else{
+                    e.target.classList.add('is-invalid');
+                    e.target.classList.remove('is-valid');
+                }
+            }
+            let onGenderChange= (e)=>{
+                let val = e.target.value;
+                let valid = true;
+                this.props.dispatch(addValue('gender', val, valid));
+            }
+            let onAgeChange= (e)=>{
+                let val = e.target.value;
+                let valid = Validation.validateAge(ageField.value);
+                this.props.dispatch(addValue('age', val, valid));
+                if(valid){
+                    e.target.classList.remove('is-invalid');
+                    e.target.classList.add('is-valid');
+                } else{
+                    e.target.classList.add('is-invalid');
+                    e.target.classList.remove('is-valid');
+                }
+            }
+            let onMiddleNameChange= (e)=>{
+                let val = e.target.value;
+                let valid = Validation.validateName(middleNameField.value);
+                this.props.dispatch(addValue('middleName', val, valid));
+                if(valid){
+                    e.target.classList.remove('is-invalid');
+                    e.target.classList.add('is-valid');
+                } else{
+                    e.target.classList.add('is-invalid');
+                    e.target.classList.remove('is-valid');
+                }
+            }
+        let form = document.getElementById('registerForm');
+
+        let onSubmitForm = (e)=>{
+            let valid=[];
+
+
+
+            for (let name in this.props.user){
+                for(let isValid in this.props.user[name]){
+                    let valueValidated = this.props.user[name].isValid;
+                    if(!valueValidated){
+                        valid.push(this.props.user[name]);
+                    }
+                }
+            }
+            if(valid.length===0){
+                document.getElementById('btnGroup').classList.remove('errorMsg');
+                let user = this.props.user;
+                let self = this;
+
+                let userFormData = new FormData();
+                userFormData.append('name', user.name.value);
+                userFormData.append('surName', user.surName.value);
+                userFormData.append('email', user.email.value);
+                userFormData.append('photo', user.photo.value);
+                userFormData.append('gender', user.gender.value);
+                userFormData.append('age', user.age.value);
+                userFormData.append('middleName', user.middleName.value);
+
+                let sendUser = new XMLHttpRequest();
```

Comment:
Why do you use XMLHttpRequest here?
We have axios, jquery.ajax and bunch of other awesome tools to do the XHR